### ISSUE TITLE
cannot upload photos any longer

### ISSUE THREAD

**Issue by gilbertohasnofb** [LevPasha/Instagram-API-python#714]
After logging in, I am getting the errors:

```
Request return 429 error!
{'message': 'Please wait a few minutes before you try again.', 'status': 'fail'}
Request return 405 error!
{'message': '', 'status': 'fail'}
Request return 404 error!
Login success!
```

But the login seems successful. The problem is I am not able to upload images in neither of my accounts using `uploadPhoto`; I always get a `False` return value:
```
>>> api.uploadPhoto(filename, caption=text)
False
```

`text` is a simple string and I made sure `filename` points to an existing file, checking with `os.path.exists(filename)` which returns `True`.

After the failed upload, if I check the last json for the status, I get:

```
>>> print(api.LastJson['status'])
ok
```

This very same code was working until around 7 days ago. Any ideas what might be the problem?

**Comment by zzzrevenue** [comment]
I also can't upload photos anymore

**Comment by gilbertohasnofb** [comment]
I tried running my script again this morning and the error 429 did not come up during login, now I only got:

```
Request return 405 error!
{'message': '', 'status': 'fail'}
Request return 404 error!
Login success!
```

But unfortunately the upload still did not go through.

**Comment by Takenobou** [comment]
#705

**Comment by gilbertohasnofb** [comment]
@Takenobou Thank you for the info, I will keep waiting and trying from time to time; if I manage to upload something I will report it here and close this issue.

**Comment by mbiparva** [comment]
I have the same problem. It is almost more than a week this is happening. The problem is the API cannot login properly. If you check the API.LastResponse.text, you will get a large json containing some user information which is odd.

**Comment by roycebank** [comment]
if you upload just photos ↓
https://paste.ubuntu.com/p/mxyYmBpryK/

**Comment by gilbertohasnofb** [comment]
@mbiparva Yes, it has been a long time already... really hoping this get fixed soon.

@roycebank Thanks for the code, I really appreciate it. But for my personal case, I am also doing things other than uploading photos. My code is also running out of a Raspberry Pi with a terminal only OS, so your solution wouldn't work for me any way. But thanks once again!

**Comment by gilbertohasnofb** [comment]
I wonder if this issue is related to the changes that Instagram is making to its API (see: https://www.instagram.com/developer/changelog/). Apparently they are ceasing to support the current API in March 2020 in favour of a new one.

**Comment by kami4ka** [comment]
The same problem

**Comment by M4xiGitHub** [comment]
same here

**Comment by mbiparva** [comment]
Can others also confirm if they can log into Instagram account using this API or not? It seems the API is no longer useable due to the failure in the login step.

I still get the error once I try to login.

**Comment by pipawoz** [comment]
Can login in in some accounts, others i have a problem with the challenge that i cant solve even with manual login on the pc/phone. 
With the ok accounts, i can follow, like and comment, but no upload photos.

**Comment by mbiparva** [comment]
> Can login in in some accounts, others i have a problem with the challenge that i cant solve even with manual login on the pc/phone.
> With the ok accounts, i can follow, like and comment, but no upload photos.

I also traced the code flow and realized that login is fine but the photo upload is not. The error the send request returns is "Something is wrong. Please check your input and try again." which is reported in the PhP API at this page too:

https://github.com/mgp25/Instagram-API/issues/2971

and here:

https://github.com/mgp25/Instagram-API/issues/2972

So I guess it is something done on the Instagram side and we cannot do anything here unless we know what is wrong.

**Comment by Fulvioo** [comment]
same here

Any alternatives?

**Comment by gilbertohasnofb** [comment]
@LevPasha is there anything that can be done from your side? Has there been any changes in the Instagram API that requires some modification to your code?

**Comment by gilbertohasnofb** [comment]
These two links might also be relevant:
* https://stackoverflow.com/questions/59063122/is-https-www-instagram-com-developer-still-working
* https://www.drupal.org/project/social_auth_instagram/issues/3092340

**Comment by FourKube** [comment]
Anyone find any other solutions for the mean time? I have tried to use python & selenium with the link from @roycebank. I was able to login, but was not able to upload a photo. Any other solutions?

**Comment by roycebank** [comment]
> Anyone find any other solutions for the mean time? I have tried to use python & selenium with the link from @roycebank. I was able to login, but was not able to upload a photo. Any other solutions?

try to change driver.find_element_by_xpath value in whiles/loops.

**Comment by knaitas** [comment]
I think the endpoint is different.
Look at node.js library (which is getting updated): https://github.com/dilame/instagram-private-api/blob/e2dadb32d55b26c3c9c4d94d64eedc1d01ad73cf/src/repositories/upload.repository.ts

Lines 20 and 21: const { body } = await this.client.request.send<UploadRepositoryPhotoResponseRootObject>({
      url: `/rupload_igphoto/${name}`,

vs Python:

response = self.s.post(self.API_URL + "upload/photo/", data=m.to_string())

endpoint in nodejs library: /rupload_igphoto/ 
endpoint in python library: upload/photo/

and it is getting 404 error in python because the endpoint does not exist anymore

**Comment by mrlee321** [comment]
> 사진 만 업로드하면 ↓
> https://paste.ubuntu.com/p/mxyYmBpryK/

thank you. i have qusetiones. 

first,   Above Link sources..... Can photoAlbum upload??(able)

second  i see your git home so.....instapy-cli <== Can photoAlbum upload??(able)   

I need instagram API's upload Album function

instapy-cli's upload-stories.py Is this possible?

**Comment by eddyizm** [comment]
Code base has not been updated since Mar 26, 2018. I think this repo has been abandoned. I know Instapy has been affected by these changes as well.
